### PR TITLE
feat(agw): create new base images `magma-dev`, `magma-test`, and `magma-trfserver`

### LINKED ISSUE #14080: Remove unused files from AGW VM images
The AGW VM images contain some unused files that significantly increase the image size.

There are about 600 MB of `.deb` packages in `/home/vagrant/magma/.cache/apt/focal`. The reason probably is that we symlink `/var/cache/apt/archives` to `/home/vagrant/magma/.cache/apt/focal`. During normal VM usage, `/home/vagrant/magma` is a shared folder from the host system, and the symlink ensures that the apt cache can be reused between different VMs. However it seems that this symlink is already created during the VM build. And when the `magma` repository gets mounted at `/home/vagrant/magma`, these files become inaccessible, but still contribute to the VM image's size.

There is a `/var/tmp/libprotobuf-mutator` directory with a size of 1GB. This probably contains a lot of files only required for the build of libprotobuf-mutator and not for running it. In the devcontainer we remove this directory after we run `ninja install`, but in the VM we keep it. This should probably be removed.

### LINKED ISSUE #14541: Create new VM base images
Remove the workaround from #14514 and use the linuxfoundation artifactory directly in the preburn image.
- magma-dev
- magma-test

Also: use this issue to extend https://github.com/magma/magma/wiki/Creation,-Testing,-and-Publishing-of-Magma-VM-Base-Images with more details

### PR DESCRIPTION
**Pull request by mpfirrmann**
## Summary

This PR changes several Ansible playbooks to update Magma's `dev`, `test`, and `trfserver` VMs. With these changes the new base images ...

- ... contain go 1.19.3.
- ... contain all prerequisites for the DHCP cli tool.
- ... contain Docker Compose V2.
- ... contain the LF artifactory as installation source.
- ... will execute less steps during provisioning to improve boot times.

Additionally:

- The packing scripts for the VMs are unified and the `trfserver` is cleaned up after packing now as well.
- All box caches are adapted.

Closes:
- #14541 
- #14080 

## Test Plan

locally:
- [x] Unit tests - `bazel test --cache_test_results=no //...`
- [x] Sudo tests - `./bazel/scripts/run_sudo_tests.sh`
- [x] AGW build
- [x] LTE integration tests
- [x] AGW build containerized
- [x] LTE integration tests containerized
- [x] Build Orc8r, SwaggerAPI, NMS along the [Quick Start Guide](https://magma.github.io/magma/docs/next/basics/quick_start_guide)
- [x] Federated integration tests
- [x] CWAG integration tests

CI:
- [x] [Sudo tests](https://github.com/mpfirrmann/magma/actions/runs/3808032073)
- [x] [LTE integration tests debian](https://github.com/mpfirrmann/magma/actions/runs/3821672324)
    - 1st run: ` s1aptests/test_attach_detach_flaky_retry_success.py` failed with a segmentation fault
- [x] LTE integration tests containerized
    - [x] [precommit](https://github.com/mpfirrmann/magma/actions/runs/3808041756)
    - [x] [extended](https://github.com/mpfirrmann/magma/actions/runs/3808042409)
- [x] [Federated integration tests](https://github.com/mpfirrmann/magma/actions/runs/3808033113)
- [x] [CWAG integration tests](https://github.com/mpfirrmann/magma/actions/runs/3808032648)

## Additional Information

- The sudo tests are quite flaky (also in CI on master) but they passed eventually. The culprits are `/pipelined/tests/test_inout_non_nat.py`'s `testFlowVlanSnapshotMatch_static1` and `testFlowVlanSnapshotMatch_static2` tests.
- :warning: #14673 had to be reverted for the LTE integ tests to all pass locally. This needs to be monitored in CI.

### CODE REVIEW

File: .github/workflows/build_all.yml [hunk 1]
```diff
@@ -117,7 +117,7 @@ jobs:
         uses: actions/cache@9b0c1fce7a93df8e3bb8926b0d6e9d89e92f20a7 # pin@v3.0.11
         with:
           path: ~/.vagrant.d/boxes/magmacore-VAGRANTSLASH-magma_dev
-          key: vagrant-box-magma-dev-v1.2.20221012
+          key: vagrant-box-magma-dev-v1.2.20221230
```
Comment (LKreutzer) [inline .github/workflows/build_all.yml]: Would it make sense to switch to `vagrant-box-magma-dev-v1.3.20221230` for the key? Same for the others.

File: lte/gateway/Vagrantfile [hunk 2]
```diff
@@ -28,7 +28,7 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
 
   config.vm.define :magma, primary: true do |magma|
     magma.vm.box = "magmacore/magma_dev"
-    magma.vm.box_version = "1.2.20221012"
```
Comment (LKreutzer) [inline lte/gateway/Vagrantfile:31]: Same as in the workflows: I think it would be reasonable to change the version to `1.3` since there are actual changes to the base image.

File: lte/gateway/deploy/roles/magma/tasks/main.yml [hunk 3]
```diff
@@ -134,73 +135,79 @@
...
 - name: Copy Envoy configuration script
-  copy: src=configure_envoy_namespace.sh dest=/usr/local/bin/configure_envoy_namespace.sh
+  copy:
+    src: configure_envoy_namespace.sh
+    dest: /usr/local/bin/configure_envoy_namespace.sh
   when: full_provision
 
-- name: Create symlink for sctpd binary
-  file: src='{{ c_build }}/sctpd/src/sctpd' path=/usr/local/sbin/sctpd state=link force=yes follow=no
+- name: Create symlink binaries
```
Comment (LKreutzer) [inline lte/gateway/deploy/roles/magma/tasks/main.yml]: Nit: `Create symlinks for binaries` or `for services`

File: lte/gateway/deploy/roles/magma/tasks/main.yml [hunk 4]
```diff
@@ -134,73 +135,79 @@
...
     path: '{{ magma_deps }}/'
     state: directory
-  when: full_provision
+  when: preburn
 
 - name: Enable IP forwarding
-  sysctl: name="net.ipv4.ip_forward" value=1 sysctl_set=yes state=present reload=yes
+  sysctl:
+    name: net.ipv4.ip_forward
+    value: '1'
```
Comment (LKreutzer) [inline lte/gateway/deploy/roles/magma/tasks/main.yml:207]: Nit: Are the quotes needed here?

File: lte/gateway/deploy/roles/magma/tasks/main.yml [hunk 5]
```diff
@@ -355,55 +355,58 @@
...
+    - /usr/local/bin/go/
+    - $(go env GOPATH)/bin/
   when: full_provision
 
 - name: Install dnsmasq
-  apt: pkg=dnsmasq state=present update_cache=yes
+  apt:
+    state: present
+    update_cache: yes
+    name: dnsmasq
```
Comment (LKreutzer) [inline lte/gateway/deploy/roles/magma/tasks/main.yml]: Nit: Could we sort `name:` to the top?

File: lte/gateway/deploy/roles/magma/tasks/main.yml [hunk 6]
```diff
@@ -355,55 +355,58 @@
...
+    state: stopped
+    enabled: no
   when: preburn
 
 - name: Install lighttpd
-  apt: pkg=lighttpd state=present update_cache=yes
+  apt:
+    state: present
+    update_cache: yes
+    name: lighttpd
```
Comment (LKreutzer) [inline lte/gateway/deploy/roles/magma/tasks/main.yml]: Same as above

File: lte/gateway/deploy/roles/magma_test/tasks/main.yml [hunk 7]
```diff
@@ -87,7 +85,10 @@
 
 - name: Install MySQL client for OAI upstreaming
   become: yes
-  apt: pkg=default-libmysqlclient-dev state=present update_cache=yes
+  apt:
+    state: present
+    update_cache: yes
+    name: default-libmysqlclient-dev
```
Comment (LKreutzer) [inline lte/gateway/deploy/roles/magma_test/tasks/main.yml]: Nit: Could we sort the name to the top?

File: lte/gateway/deploy/roles/trfserver/tasks/main.yml [hunk 8]
```diff
@@ -76,11 +76,13 @@
       - git+https://github.com/thiezn/iperf3-python#egg=iperf3-python
   when: preburn
 
-- name: remove preinstalled iperf3 and libiperf0
+- name: Remove preinstalled iperf3 and libiperf0
   apt:
-    name: ['iperf3', 'libiperf0']
     state: absent
     update_cache: yes
+    name:
```
Comment (LKreutzer) [inline lte/gateway/deploy/roles/trfserver/tasks/main.yml:81]: Nit: Sort to the top

File: orc8r/tools/ansible/roles/docker/tasks/install_debian.yml [hunk 9]
```diff
@@ -15,57 +15,11 @@
...
-  ignore_errors: true # for the cwag dev VM
-
 - name: Install packages to allow apt to use a repository over HTTPS
   apt:
-    name: "{{ packages }}"
     state: present
     update_cache: true
-  vars:
-    packages:
+    name:
```
Comment (LKreutzer) [inline orc8r/tools/ansible/roles/docker/tasks/install_debian.yml:20]: Nit: sort to top

File: orc8r/tools/ansible/roles/docker/tasks/install_debian.yml [hunk 10]
```diff
@@ -79,28 +33,21 @@
...
     state: present
 
 - name: Install the latest version of Docker CE
   apt:
-    name: "{{ packages }}"
     state: present
     update_cache: true
-  vars:
-    packages:
+    name:
```
Comment (LKreutzer) [inline orc8r/tools/ansible/roles/docker/tasks/install_debian.yml:48]: Nit: Sort to top

File: orc8r/tools/ansible/roles/python_dev/tasks/main.yml [hunk 11]
```diff
@@ -51,18 +51,14 @@
 # Add common convenience aliases
 #################################
 
-- name: Set a convenience function for activating the virtualenv
+- name: Set a convenience functions for virtualenv
```
Comment (LKreutzer) [inline orc8r/tools/ansible/roles/python_dev/tasks/main.yml]: Nit: `Set convenience functions for virtualenv`

File: orc8r/tools/packer/magma-dev-virtualbox.json [hunk 12]
```diff
@@ -51,7 +51,6 @@
           "4"
         ]
       ],
-      "name": "magma",
```
Comment (LKreutzer) [inline orc8r/tools/packer/magma-dev-virtualbox.json:54]: Why is this removed?

File: orc8r/tools/packer/magma-test-virtualbox.json [hunk 13]
```diff
@@ -82,10 +81,7 @@
       "expect_disconnect": true
     },
     {
-      "extra_arguments": [
```
Comment (LKreutzer) [inline orc8r/tools/packer/magma-test-virtualbox.json:85]: Is this change done with a formatter?

**Review by LKreutzer — CHANGES_REQUESTED** [review]
Did you run a formatter on the `yml` and `json` files? 
In particular the sorting of the `name` attribute should be at the top imo - I stopped commenting this point half-way through because it is frequent. If this was done by a formatter and follows some convention ignore the sorting comments.

**Comment by mpfirrmann** [comment]
> Did you run a formatter on the `yml` and `json` files? In particular the sorting of the `name` attribute should be at the top imo - I stopped commenting this point half-way through because it is frequent. If this was done by a formatter and follows some convention ignore the sorting comments.

I did not use a formatter for these files. The "name not on top" style was historically used all over the repo, so I just stuck with that. I changed it for the files I touched in this PR.

File: orc8r/tools/packer/magma-dev-virtualbox.json (hunk 12 above)
Comment (mpfirrmann) [reply orc8r/tools/packer/magma-dev-virtualbox.json:54]: Because this parameter doesn't do anything. Also remove in the trfserver file.

File: orc8r/tools/packer/magma-test-virtualbox.json (hunk 13 above)
Comment (mpfirrmann) [reply orc8r/tools/packer/magma-test-virtualbox.json:85]: All VM `.json`s are now formatted the same. The order is debatable and was also topic during the last PR releasing base images.

File: .github/workflows/build_all.yml (hunk 1 above)
Comment (mpfirrmann) [reply .github/workflows/build_all.yml]: I was not part of the `v1.1` -> `v1.2` discussion, so I can't comment on any real convention. Previously, we just adapted the timestamp.

File: lte/gateway/deploy/roles/magma/tasks/main.yml (hunk 4 above)
Comment (mpfirrmann) [reply lte/gateway/deploy/roles/magma/tasks/main.yml:207]: They are present in the [Ansible docs](https://docs.ansible.com/ansible/latest/collections/ansible/posix/sysctl_module.html).

File: .github/workflows/build_all.yml (hunk 1 above)
Comment (LKreutzer) [reply .github/workflows/build_all.yml]: This is perhaps a point to be raised in a larger round then. I would suggest to move to a common version for the new VMs.

File: lte/gateway/deploy/roles/magma/tasks/main.yml (hunk 4 above)
Comment (LKreutzer) [reply lte/gateway/deploy/roles/magma/tasks/main.yml:207]: Okay - just seems curious given that we do not use quotes on the other integer values.

**Review by LKreutzer — COMMENTED** [review]
Other than the comments about the versions, lgtm. Please re-request a review once that has been addressed.

File: .github/workflows/build_all.yml (hunk 1 above)
Comment (mpfirrmann) [reply .github/workflows/build_all.yml]: The versions are updated to 1.3 in the cloud and the PR.

**Review by LKreutzer — APPROVED** [review]

**Review by Neudrino — APPROVED** [review]

File: orc8r/tools/ansible/roles/python_dev/tasks/main.yml [hunk 14]
```diff
@@ -51,18 +51,14 @@
 # Add common convenience aliases
 #################################
 
-- name: Set a convenience function for activating the virtualenv
+- name: Set functions for virtualenv
```
Comment (voisey) [inline orc8r/tools/ansible/roles/python_dev/tasks/main.yml]: Nit: I would say it's easier to understand what's going on here at a glance with "convenience functions" in the name rather than just "functions"

File: orc8r/tools/packer/scripts/ubuntu_setup.sh [hunk 15]
```diff
@@ -25,9 +25,9 @@ add-apt-repository -y ppa:ansible/ansible
 apt-get update
 apt-get install -y ansible
 
-# Mark ca-certificates-java to not have it removed via autoremove;
+# Install ca-certificates-java;
 # needed for swagger_codegen_jar
-apt-mark manual ca-certificates-java 
+apt-get install -y ca-certificates-java
```
Comment (voisey) [inline orc8r/tools/packer/scripts/ubuntu_setup.sh:30]: What's the reason for the change here?

File: orc8r/tools/ansible/roles/docker/tasks/install_debian.yml [hunk 16]
```diff
@@ -17,35 +17,38 @@
...
     dest: /tmp/docker.key
 
-- name: Add Docker’s official GPG key
+- name: Add Docker's official GPG key
   apt_key:
     file: /tmp/docker.key
     state: present
 
-- name: Add the stable repository for Docker
+- name: Add the focal repository for Docker
```
Comment (voisey) [inline orc8r/tools/ansible/roles/docker/tasks/install_debian.yml]: Nit: Maybe it makes more sense to keep it as "stable" so that this doesn't need to be changed if the distribution is updated

File: lte/gateway/deploy/roles/magma/tasks/main.yml [hunk 17]
```diff
@@ -134,73 +135,79 @@
...
-  when: full_provision
-
-- name: Create symlink for connectiond binary
-  file: src='{{ c_build }}/connection_tracker/src/connectiond' path=/usr/local/bin/connectiond state=link force=yes follow=no
-  when: full_provision
-
-- name: Create symlink for li agent binary
-  file: src='{{ c_build }}/li_agent/src/liagentd' path=/usr/local/bin/liagentd state=link force=yes follow=no
+  file:
+    src: '{{ magma_root }}/lte/gateway/python/dhcp_helper_cli/dhcp_helper_cli.py'
```
Comment (voisey) [inline lte/gateway/deploy/roles/magma/tasks/main.yml:168]: Just to check, was it intentional that you removed `/home/{{ ansible_user }}` from the start of the path here?

File: orc8r/tools/packer/scripts/ubuntu_setup.sh (hunk 15 above)
Comment (mpfirrmann) [reply orc8r/tools/packer/scripts/ubuntu_setup.sh:30]: The `ca-certificates-java` is not installed by default anymore, so we have to install it manually. It was installed by default previously, that's why marking it manually installed was enough. I have no idea, why this changed.

File: lte/gateway/deploy/roles/magma/tasks/main.yml (hunk 17 above)
Comment (mpfirrmann) [reply lte/gateway/deploy/roles/magma/tasks/main.yml:168]: Yes. `magma_root` is defined as `/home/{{ ansible_user }}/magma` in the `magma_$NAME.yml` of the VMs.

**Review by voisey — APPROVED** [review]

**Review by wolfseb — APPROVED** [review]
lgtm
